docs/topics/control-flow.md

```diff
@@ -177,6 +177,82 @@ fun Request.getBody() =
+
+Guard conditions allow you to include more than one condition to the branches of a `when` expression, making complex control flow more explicit and concise.
+
+To include a guard condition in a branch, place the guard condition after the primary condition, separated by `if`:
+
+```kotlin
+sealed interface Animal {
+    data class Cat(val mouseHunter: Boolean) : Animal
+    data class Dog(val breed: String) : Animal
+}
+
+fun feedAnimal(animal: Animal) {
+    when (animal) {
+        // Branch with only primary condition. Returns `feedDog()` when `Animal` is `Dog`
+        is Animal.Dog -> feedDog()
+        // Branch with both primary and guard conditions. Returns `feedCat()` when `Animal` is `Cat` and is not `mouseHunter`
+        is Animal.Cat if !animal.mouseHunter -> feedCat()
+        // Returns "Unknown animal" if none of the above conditions match
+        else -> println("Unknown animal")
+    }
+}
+```
+
+In a single `when` expression, you can combine branches with and without guard conditions. 
+The code in a branch with a guard condition runs only if both the primary condition and the guard condition evaluate to true.
+If the primary condition does not match, the guard condition is not evaluated. 
+
+If you use guard conditions in `when` statements without an `else` branch, if none of the conditions matches, none of the branches is executed. 
+
+Otherwise, if you use guard conditions in `when` expressions without an `else` branch, the compiler requires you to declare all the possible cases (to avoid runtime errors).
+
+Additionally, guard conditions support `else if`:
+
+```kotlin
+when (animal) {
+    // Checks if `animal` is `Dog`
+    is Animal.Dog -> feedDog()
+    // Guard condition that checks if `animal` is `Cat` and not `mouseHunter`
+    is Animal.Cat if !animal.mouseHunter -> feedCat()
+    // Returns giveLettuce() if none of the above conditions match and animal.eatsPlants is true
+    else if animal.eatsPlants -> giveLettuce()
+    // Returns "Unknown animal" if none of the above conditions match
+    else -> println("Unknown animal")
+}
+```
+
+Combine multiple guard conditions within a single branch using the boolean operators `&&` (AND) or `||` (OR).
+It is [strongly recommended](coding-conventions.md#guard-conditions-in-when-expression) to use parentheses around the boolean expressions to avoid confusion:
+
+```kotlin
+when (animal) {
+    is Animal.Cat if (!animal.mouseHunter && animal.hungry) -> feedCat()
+}
+```
+
+You can use guard conditions in any `when` expression or statement with a subject, except the case when you have multiple conditions separated by a comma
+(for example: `0, 1 -> print("x == 0 or x == 1")`).
+
+> To enable guard conditions in CLI, run the following command:
+>
+> `kotlinc -Xwhen-guards main.kt`
+>
+> To enable guard conditions in Gradle, run the following command:
+>
+> `kotlin.compilerOptions.freeCompilerArgs.add("-Xwhen-guards")t`
```

Comment:
Shouldn't the option be added to the `build.gradle.kts` file?

Reply:
True. Great catch!